### PR TITLE
Add keyboard commands to index route. Closes #18.

### LINKED ISSUE #18: add keyboard commands to index route
* up/down will select new item in search results
* enter will navigate to URL of selected app
* more?

### PR DESCRIPTION
**Pull request by cowboy**
Also:

* Fixed sorting!

Lots of refactoring:

* Only display Loading component in Container components
* Don't populate state with props
* Remove redundant code
* Prefer functional components over classical components
* Use destructuring assignment whenever possible
* Better define propTypes for nested objects (params)

TODO: #54, once the page is a little more styled.

### CODE REVIEW

File: src/components/search/index.jsx [hunk 1]
```diff
@@ -28,42 +30,58 @@ export default class Search extends BoundComponent {
...
 
-  render() {
-    if (!this.props.services) {
-      return (
-        <Loading />
-      );
+  handleKeyDown(event) {
+    let args;
+    switch (event.keyCode) {
+      case 38:
```
Comment (tkellen) [inline src/components/search/index.jsx]: can we make these integers constants at the top of the file so it's a bit easier to read later?

**Comment by jennschiffer** [comment]
:+1:

File: src/components/service-list-container/index.jsx [hunk 2]
```diff
@@ -24,31 +26,58 @@ export default class ServiceListContainer extends BoundComponent {
...
+  setQueriedServices(services) {
+    const visibleServices = services || this.state.services;
+    const selectedIndex = 0;
+    this.setState({visibleServices, selectedIndex});
+  }
+
+  navigate(delta) {
+    let {selectedIndex} = this.state;
+    const {visibleServices} = this.state;
+    if (delta) {
```
Comment (tkellen) [inline src/components/service-list-container/index.jsx]: a comment here explaining what this is doing would be mad helpful for newcomers to the code (like you in 3 months!)

File: src/components/search/index.jsx (hunk 1 above)
Comment (cowboy) [inline src/components/search/index.jsx]: Makes sense.

File: src/components/service-list-container/index.jsx [hunk 3]
```diff
@@ -24,31 +26,58 @@ export default class ServiceListContainer extends BoundComponent {
...
+      selectedIndex = Math.max(0, Math.min(visibleServices.length - 1, selectedIndex + delta));
+      this.setState({selectedIndex});
+    }
+    else {
+      const service = visibleServices[selectedIndex];
```
Comment (tkellen) [inline src/components/service-list-container/index.jsx]: why the intermediate variable here?

**Comment by tkellen** [comment]
pretty rad!

File: src/components/service-list-container/index.jsx (hunk 2 above)
Comment (cowboy) [inline src/components/service-list-container/index.jsx]: What what is doing? The `delta` var?

Comment (tkellen) [inline src/components/service-list-container/index.jsx]: Whoops, I meant to put that on the line below.

File: src/components/service-list-container/index.jsx (hunk 3 above)
Comment (cowboy) [inline src/components/service-list-container/index.jsx]: leftover code, i'll remove it.

File: src/components/service-list-container/index.jsx (hunk 2 above)
Comment (cowboy) [inline src/components/service-list-container/index.jsx]: I'll add a comment, but i've done enough math in code to be able to recognize what that does

**Pull request merged**